### DISCUSSION TITLE
Allow globalSets to be merged into a single entrified section

### DISCUSSION THREAD

**Discussion by svondervoort** [craftcms/cms#14632] [Ideas] [answered]
We have this situation where we created multiple global-sets for stuff. In hindsight we would like to merge those all into a single entry with multple tabs.

Now I tried this by first entrifying one global-set and then adding the required fields of the other sets into the same `entry type`, but when running the commands to entrify the data on the staging and live environment the fields get emptied/overwritten by the next command. So only the last entrification command fields for the same entry get saved.

Example: 
```
php craft entrify/global-set globalGeneral --section=globalSettings
php craft entrify/global-set notice --section=globalSettings
php craft entrify/global-set globalBottomPageCTA --section=globalSettings
php craft entrify/global-set globalSearch --section=globalSettings
```

Only the fields from `globalSearch` are filled in after running all these command.

Is it possible to maintain the field values so only the corresponding fields get merged?

**Comment by brandonkelly** [comment, marked as answer]
There’s no way to do this with the built-in CLI commands, and it would be tricky to automate since you could be reusing the same field across multiple global sets. (Craft 5 adds multi-instance support, but that’s not available to all fields, so still not a perfect solution.)

If you really want to automate it, you could use a content migration. Run `php craft migrate/create populate_global_settings` to generate the migration, and then fill its `safeUp()` method up with something like:

```php
use Craft;
use craft\elements\Entry;

public function safeUp(): bool
{
    $entry = Entry::find()->section('globalSettings')->one();
    if (!$entry) {
        echo "globalSettings entry not found\n";
    }

    $globalSetHandles = [
        'globalGeneral',
        'notice',
        'globalBottomPageCTA',
        'globalSearch',
    ];

    foreach ($globalSetHandles as $handle) {
        $globalSet = Craft::$app->getGlobals()->getSetByHandle($handle, withTrashed: true);
        if (!$globalSet) {
            echo "Global set $handle not found\n";
            return false;
        }
        $entry->setFieldValues($globalSet->getFieldValues());
    }

    if (!Craft::$app->getElements()->saveElement($entry, false)) {
        echo "globalSettings entry not saved\n";
        return false;
    }

    return true;
}
```

**Comment by svondervoort** [comment]
I didn't even think about the reusable fields. You are right, it was worth a shot.
